dashboard/lib/sign_up_tracking.rb

```diff
@@ -1,11 +1,38 @@
 require 'cdo/firehose'
+require 'dynamic_config/dcdo'
 
 module SignUpTracking
-  def self.begin_sign_up_tracking(session)
+  NOT_IN_STUDY_GROUP = 'v1'
+  CONTROL_GROUP = 'v2-control'
+  NEW_SIGN_UP_GROUP = 'v2-finish-sign-up'
+
+  def self.study_group(session)
+    session[:sign_up_study_group] || NOT_IN_STUDY_GROUP
+  end
+
+  def self.new_sign_up_experience?(session)
+    study_group(session) == NEW_SIGN_UP_GROUP
+  end
+
+  def self.begin_sign_up_tracking(session, split_test: false)
     unless session[:sign_up_tracking_expiration]&.future?
       session[:sign_up_uid] = SecureRandom.uuid.to_s
       session[:sign_up_tracking_expiration] = 5.minutes.from_now
     end
+
+    if split_test
+      session[:sign_up_study_group] = Random.rand(100) < split_test_percentage ?
+          NEW_SIGN_UP_GROUP : CONTROL_GROUP
+    end
+  end
```

Comment:
The `begin_sign_up_tracking` method may be called multiple times in a sign-up flow.  We only want to set a uid and expiration the _first_ time this happens, but sometimes we'll only know the provider (and therefore be able to opt-in to the split test) on a subsequent call.